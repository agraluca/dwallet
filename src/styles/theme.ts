export default {
  grid: {
    container: "126rem",
    gutter: "3.2rem",
  },
  border: {
    radius: "0.8rem",
  },
  font: {
    family:
      "'Big Shoulders Display', -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif",
    weight: {
      thin: 400,
      normal: 500,
      bold: 700,
    },
    sizes: {
      xsmall: "1.2rem",
      small: "1.8rem",
      medium: "2.2rem",
      large: "3.2rem",
      xlarge: "3.6rem",
    },
  },
  colors: {
    yellow: "#F5CB5C",
    black: "#242423",
    lightBlack: "#333333",
    blue: "#CFDBD5",
    lightBlue: "#E8EDDF",
    green: "#33CC95",
    red: "#E52E4D",
    white: "#FAFAFA",
  },
  spacings: {
    xxsmall: "0.8rem",
    xsmall: "1.6rem",
    small: "2.4rem",
    medium: "3.2rem",
    large: "4.0rem",
    xlarge: "4.8rem",
    xxlarge: "5.6rem",
  },
  layers: {
    base: 10,
    menu: 20,
    overlay: 30,
    modal: 40,
    alwaysOnTop: 50,
  },
} as const;
