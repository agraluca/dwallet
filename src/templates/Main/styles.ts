import styled, { css } from "styled-components";
import media from "styled-media-query";

export const Wrapper = styled.main`
  ${({ theme }) => css`
    background-color: ${theme.colors.black};
    width: 100%;
    height: 100%;
    padding: ${theme.spacings.medium};
    display: flex;
    justify-content: center;
    align-items: center;
    color: ${theme.colors.white};
  `}
`;

export const Logo = styled.img`
  width: 10rem;
`;

export const TitleWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;

    .login--title {
      font-size: calc(${theme.font.sizes.large} * 2);
    }
  `}
`;

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.large};
    font-weight: ${theme.font.weight.thin};
    line-height: calc(${theme.font.sizes.large} * 1.2);
    max-width: 40rem;
  `}
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;

  ${media.lessThan("large")`
    justify-content: center;
    align-items: center;
  `}
`;

export const IlustrationWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const Illustration = styled.img`
  width: min(30rem, 100%);

  ${media.lessThan("large")`
    display: none;
  `}
`;

export const WaveIllustration = styled.img`
  position: absolute;
  bottom: 0;
  left: 0;
  width: 100vw;
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 1rem;
    width: 40rem;
    border-radius: ${theme.border.radius};
    padding: ${theme.spacings.xsmall};
    font-size: ${theme.font.sizes.small};
    font-weight: ${theme.font.weight.normal};
    cursor: pointer;
    transition: filter 0.2s;
    margin-top: ${theme.spacings.small};
    border: 0.2rem solid ${theme.colors.white};
    color: ${theme.colors.white};
    background-color: ${theme.colors.black};

    .login--google-logo {
      width: 2.8rem;
      height: 2.8rem;
    }

    &:hover {
      filter: brightness(0.8);
    }

    ${media.lessThan("medium")`
      width: 100%;
   `}
  `}
`;
