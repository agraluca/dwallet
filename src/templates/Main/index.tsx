import { useState } from "react";

import Wrapper from "components/Wrapper";
import Heading from "components/Heading";
import Input from "components/Input";
import { Button } from "components/Button";
import Link from "next/link";

import useAuth from "hooks/useAuth";

import * as S from "./styles";

import { useAppSelector } from "hooks/useReduxHooks";

function Main({
  title = "DWallet",
  description = "Seu sistema de balanceamento de carteira",
}) {
  const [formValues, setFormValues] = useState({
    email: "",
    password: "",
  });
  const { signIn } = useAuth();

  const { getTokenLoading } = useAppSelector((state) => state.loading);

  function handleInput(field: string, value: string) {
    setFormValues((prevState) => ({ ...prevState, [field]: value }));
  }

  async function handleSubmit(e: React.FormEvent) {
    e.preventDefault();
    signIn(formValues);
  }

  return (
    <Wrapper zeroIndex>
      <S.Wrapper>
        <S.Content>
          <S.TitleWrapper>
            <S.Logo src="/img/logo.svg" alt="Logo do projeto" />
            <Heading level={1} className="login--title">
              {title}
            </Heading>
          </S.TitleWrapper>
          <S.Description>{description}</S.Description>

          <S.FormLogin onSubmit={handleSubmit}>
            <Input
              type="text"
              onInputChange={(value) => handleInput("email", value)}
              placeholder="Email"
              inputSize="full"
              icon="email"
            />
            <Input
              type="password"
              onInputChange={(value) => handleInput("password", value)}
              placeholder="Senha"
              inputSize="full"
              icon="lock"
            />
            <Button
              className="submitButton"
              type="submit"
              disabled={getTokenLoading}
            >
              {getTokenLoading ? <S.FormLoading /> : <span>Entrar</span>}
            </Button>
          </S.FormLogin>
          <Link href="/register">
            <S.RegisterLink>Crie sua conta</S.RegisterLink>
          </Link>
        </S.Content>
        <S.IlustrationWrapper>
          <S.Illustration src="/img/hero.svg" alt="Uma carteira" />
        </S.IlustrationWrapper>
      </S.Wrapper>
    </Wrapper>
  );
}

export default Main;
