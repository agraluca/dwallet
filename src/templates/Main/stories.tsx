import { Story, Meta } from "@storybook/react/types-6-0";

import Main from ".";

export default {
  title: "Main",
  component: Main,
  // opcinal
  args: {
    title: "DWallet",
    description: "Seu sistema de balanceamento de carteira",
  },
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;
