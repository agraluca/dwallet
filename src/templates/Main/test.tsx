import { screen } from "@testing-library/react";
import theme from "styles/theme";
import { renderWithTheme } from "utils/tests/helpers";

import Main from ".";

describe("<Main />", () => {
  it("should render the heading", () => {
    const { container } = renderWithTheme(<Main />);

    expect(
      screen.getByRole("heading", { name: /DWallet/i })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the colors correctly", () => {
    const { container } = renderWithTheme(<Main />);
    expect(container.firstChild).toHaveStyle({
      "background-color": theme.colors.black,
    });
  });

  it("should render the button correctly", () => {
    renderWithTheme(<Main />);
    const button = screen.getByText(/Entrar com o Google/i);
    expect(button).toBeInTheDocument();
  });
});
