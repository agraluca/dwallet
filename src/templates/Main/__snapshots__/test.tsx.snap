// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c2 {
  background-color: #242423;
  width: 100%;
  height: 100%;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FAFAFA;
}

.c5 {
  width: 10rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 .login--title {
  font-size: calc(3.2rem * 2);
}

.c7 {
  font-size: 3.2rem;
  font-weight: 400;
  line-height: calc(3.2rem * 1.2);
  max-width: 40rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 50%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 50%;
}

.c10 {
  width: min(30rem,100%);
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1rem;
  width: 40rem;
  border-radius: 0.8rem;
  padding: 1.6rem;
  font-size: 1.8rem;
  font-weight: 500;
  cursor: pointer;
  -webkit-transition: filter 0.2s;
  transition: filter 0.2s;
  margin-top: 2.4rem;
  border: 0.2rem solid #FAFAFA;
  color: #FAFAFA;
  background-color: #242423;
}

.c8 .login--google-logo {
  width: 2.8rem;
  height: 2.8rem;
}

.c8:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
}

.c0 {
  background-color: #242423;
  position: relative;
  width: 100%;
  height: 100%;
}

.c11 {
  position: fixed;
  bottom: 0;
  width: 100%;
  pointer-events: none;
}

.c1 {
  position: absolute;
  max-width: 126rem;
  min-width: 30rem;
  margin: 0 auto;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 10;
}

.c6 {
  color: #FAFAFA;
}

@media (max-width:1170px) {
  .c3 {
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    width: 100%;
  }
}

@media (max-width:1170px) {
  .c9 {
    display: none;
  }
}

@media (max-width:450px) {
  .c8 {
    width: 100%;
  }
}

<section
  class="c0"
>
  <section
    class="c1"
  >
    <main
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <img
            alt="Logo do projeto"
            class="c5"
            src="/img/logo.svg"
          />
          <h1
            class="c6 login--title"
            color="white"
          >
            DWallet
          </h1>
        </div>
        <p
          class="c7"
        >
          Seu sistema de balanceamento de carteira
        </p>
        <button
          class="c8"
        >
          <img
            alt="google logo"
            class="login--google-logo"
            src="/icons/google.svg"
          />
          Entrar com o Google
        </button>
      </div>
      <div
        class="c9"
      >
        <img
          alt="Uma carteira"
          class="c10"
          src="/img/hero.svg"
        />
      </div>
    </main>
  </section>
  <img
    alt="wave"
    class="c11 wave"
    src="/img/wave.svg"
  />
</section>
`;
